### PR TITLE
Chronos: Add Keras models for LSTM and Seq2Seq

### PR DESCRIPTION
**Pull request by liangs6212**
List of changes:
 * add keras models/ut for LSTM and Seq2Seq
 * split test env for tf1 and tf2
   * Apptest, including deprecated, network_traffic_model_forecaster(mtnet)
   * UT, including time_sequence, tf1 model, tf1 ut

### CODE REVIEW

**Comment by liangs6212** [comment]
related issue: #4023

**Comment by liangs6212** [comment]
list: 
 - [ x ] : lstm/seq2seq model
 - [ ] : BASETF2Forecaster

**Comment by TheaperDeng** [comment]
> related issue: #4023 list:
> 
>     * [x]  : lstm/seq2seq model
> 
>     * [x]  : rename seq2seq.py/VanillaLSTM to seq2seq_keras.py/VanillaLSTM_keras.py
> 
>     * [ ]  : BASETF2Forecaster

We don't need basetf2forecaster in this PR since that will make this PR too large to be managed.

**Comment by liangs6212** [comment]
ok, when BASETF2Forecaster is developed, I will split this pr.

File: python/chronos/src/bigdl/chronos/model/Seq2Seq_keras.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+from bigdl.nano.tf.keras import Model
+from keras.layers import LSTM, Dense, Lambda, Reshape
+
+
+class LSTMSeq2Seq(Model):
+    def __init__(self,
+                 input_feature_num,
+                 future_seq_len,
+                 output_feature_num,
+                 lstm_hidden_dim,
```
Comment (TheaperDeng) [inline python/chronos/src/bigdl/chronos/model/Seq2Seq_keras.py:28]: give some default value as in Seq2Seq_pytorch.py

File: python/chronos/src/bigdl/chronos/autots/deprecated/model/time_sequence.py [hunk 2]
```diff
@@ -16,8 +16,8 @@
 from copy import deepcopy
 
 from bigdl.orca.automl.model.abstract import BaseModel, ModelBuilder
-from bigdl.chronos.model.VanillaLSTM import VanillaLSTM
-from bigdl.chronos.model.Seq2Seq import LSTMSeq2Seq
+from bigdl.chronos.model.tf1_model._VanillaLSTM_keras import VanillaLSTM
+from bigdl.chronos.model.tf1_model._Seq2Seq_kears import LSTMSeq2Seq
```
Comment (TheaperDeng) [inline python/chronos/src/bigdl/chronos/autots/deprecated/model/time_sequence.py:20]: maybe we can change the filename layout to this
```python
bigdl.chronos.model.{tf1, tf2}.VanillaLSTM
bigdl.chronos.model.{tf1, tf2}.Seq2Seq
```

**Review by TheaperDeng — APPROVED** [review]
LGTM, pls run jenkins

File: python/chronos/test/bigdl/chronos/model/tf2/test_VanillaLSTM_keras.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        checkpoint_file = tempfile.TemporaryDirectory().name
+        self.model.fit(self.train_data[0],
+                       self.train_data[1],
+                       epochs=2,
+                       validation_data=self.val_data)
+        self.model.save(checkpoint_file)
+        restore_model = keras.models.load_model(checkpoint_file, custom_objects={"LSTMModel": LSTMModel})
+        model_res = self.model.evaluate(self.test_data[0], self.test_data[1])
+        restore_model_res = restore_model.evaluate(self.test_data[0], self.test_data[1])
+        assert isinstance(restore_model, Model)
```
Comment (TheaperDeng) [inline python/chronos/test/bigdl/chronos/model/tf2/test_VanillaLSTM_keras.py:72]: is this restore model a `LSTMModel`?

File: python/chronos/test/bigdl/chronos/model/tf2/test_Seq2Seq_keras.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        checkpoint_file = tempfile.TemporaryDirectory().name
+        self.model.fit(self.train_data[0],
+                       self.train_data[1],
+                       epochs=2,
+                       validation_data=self.test_data)
+        self.model.save(checkpoint_file)
+        restore_model = keras.models.load_model(checkpoint_file, custom_objects={"LSTMSeq2Seq": LSTMSeq2Seq})
+        model_res = self.model.evaluate(self.test_data[0], self.test_data[1])
+        restore_model_res = restore_model.evaluate(self.test_data[0], self.test_data[1])
+        assert isinstance(restore_model, Model)
```
Comment (TheaperDeng) [inline python/chronos/test/bigdl/chronos/model/tf2/test_Seq2Seq_keras.py:74]: is this restore model a `LSTMSeq2Seq`?

**Comment by liangs6212** [comment]
http://10.112.231.51:18889/job/BigDL-Chronos-PR-Validation-Env-Test/24/

File: python/chronos/test/bigdl/chronos/model/tf2/test_VanillaLSTM_keras.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        y = np.random.randn(num_samples, output_time_steps, output_feature_dim)
+        return x, y
+
+    train_data = get_x_y(num_train_samples)
+    val_data = get_x_y(num_val_samples)
+    test_data = get_x_y(num_test_samples)
+    return train_data, val_data, test_data
+
+
+class TestVanillaLSTM(TestCase):
```
Comment (TheaperDeng) [inline python/chronos/test/bigdl/chronos/model/tf2/test_VanillaLSTM_keras.py:46]: should make these test only available under a tf2 env

File: python/chronos/test/bigdl/chronos/model/tf2/test_Seq2Seq_keras.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    def get_x_y(num_samples):
+        x = np.random.randn(num_samples, past_seq_len, input_feature_num)
+        y = np.random.randn(num_samples, future_seq_len, output_feature_num)
+        return x, y
+    train_data = get_x_y(train_num_samples)
+    test_data = get_x_y(test_num_samples)
+    return train_data, test_data
+
+
+class TestSeq2Seq(ZooTestCase):
```
Comment (TheaperDeng) [inline python/chronos/test/bigdl/chronos/model/tf2/test_Seq2Seq_keras.py:44]: should make these test only available under a tf2 env

**Review by TheaperDeng — APPROVED** [review]

**Comment by liangs6212** [comment]
http://10.112.231.51:18889/job/BigDL-Chronos-NB-Validation/174/
http://10.112.231.51:18889/job/Chronos-PR-Validation-Spark-2.4-CI/24/
http://10.112.231.51:18889/job/Chronos-PR-Validation-Spark-3.1-CI/27/

**Comment by liangs6212** [comment]
http://10.112.231.51:18888/job/BigDL-PRVN-chronos-Python-Spark-2.4-py37-ray-part1/529/
http://10.112.231.51:18888/job/BigDL-PRVN-chronos-Python-Spark-2.4-py37-ray-part2/482/
http://10.112.231.51:18888/job/BigDL-PRVN-chronos-Python-Spark-2.4-py36-ray-part2/473/
http://10.112.231.51:18888/job/BigDL-Chronos-PR-Validation/488/

**Pull request merged**